src/jit/hwintrinsicxarch.cpp

```diff
@@ -405,13 +405,84 @@ GenTree* Compiler::impSSEIntrinsic(NamedIntrinsic intrinsic, CORINFO_METHOD_HAND
     GenTree* retNode = nullptr;
     GenTree* op1     = nullptr;
     GenTree* op2     = nullptr;
+    GenTree* op3     = nullptr;
+    GenTree* op4     = nullptr;
+
     switch (intrinsic)
     {
+        case NI_SSE_Set:
+            assert(sig->numArgs == 4);
+            op4     = impSIMDPopStack(TYP_FLOAT);
+            op3     = impSIMDPopStack(TYP_FLOAT);
+            op2     = impSIMDPopStack(TYP_FLOAT);
+            op1     = impSIMDPopStack(TYP_FLOAT);
+            retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, op2, op3, op4, intrinsic, TYP_FLOAT, 16);
+            break;
+
+        case NI_SSE_Shuffle:
+            assert(sig->numArgs == 3);
+            op3     = impSIMDPopStack(TYP_INT);
+            op2     = impSIMDPopStack(TYP_SIMD16);
+            op1     = impSIMDPopStack(TYP_SIMD16);
+            retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, op2, op3, intrinsic, TYP_FLOAT, 16);
+            break;
+
         case NI_SSE_Add:
+        case NI_SSE_And:
+        case NI_SSE_AndNot:
+        case NI_SSE_CompareEqual:
+        case NI_SSE_CompareGreaterThan:
+        case NI_SSE_CompareGreaterThanOrEqual:
+        case NI_SSE_CompareLessThan:
+        case NI_SSE_CompareLessThanOrEqual:
+        case NI_SSE_CompareNotEqual:
+        case NI_SSE_CompareNotGreaterThan:
+        case NI_SSE_CompareNotGreaterThanOrEqual:
+        case NI_SSE_CompareNotLessThan:
+        case NI_SSE_CompareNotLessThanOrEqual:
+        case NI_SSE_CompareOrdered:
+        case NI_SSE_CompareUnordered:
+        case NI_SSE_Divide:
+        case NI_SSE_Max:
+        case NI_SSE_Min:
+        case NI_SSE_MoveHighToLow:
+        case NI_SSE_MoveLowToHigh:
+        case NI_SSE_Multiply:
+        case NI_SSE_Or:
+        case NI_SSE_Subtract:
+        case NI_SSE_UnpackHigh:
+        case NI_SSE_UnpackLow:
+        case NI_SSE_Xor:
             assert(sig->numArgs == 2);
             op2     = impSIMDPopStack(TYP_SIMD16);
             op1     = impSIMDPopStack(TYP_SIMD16);
-            retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, op2, NI_SSE_Add, TYP_FLOAT, 16);
+            retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, op2, intrinsic, TYP_FLOAT, 16);
+            break;
+
+        case NI_SSE_StaticCast:
+            assert(sig->numArgs == 1);
+            op1 = impSIMDPopStack(TYP_SIMD16);
+            retNode =
+                gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, intrinsic, getBaseTypeOfSIMDType(sig->retTypeSigClass), 16);
```

Comment:
Generic HW intrinsics must return `nullptr` if the type arguments are non-numeric types.

Reply:
Will fix.

Reply:
Fixed.